record_linux/lib/record_linux.dart

```diff
@@ -129,13 +130,79 @@ class RecordLinux extends RecordPlatform {
     );
   }
 
+  @override
+  Future<Stream<Uint8List>> startStream(
+      String recorderId, RecordConfig config) async {
+    final supported = await isEncoderSupported(recorderId, config.encoder);
+    if (!supported) {
+      throw Exception('${config.encoder} is not supported.');
+    }
+
+    String numChannels;
+    if (config.numChannels == 6) {
+      numChannels = '5.1';
+    } else if (config.numChannels == 8) {
+      numChannels = '7.1';
+    } else if (config.numChannels == 1 || config.numChannels == 2) {
+      numChannels = config.numChannels.toString();
+    } else {
+      throw Exception('${config.numChannels} config is not supported.');
+    }
+
+    final streamController = StreamController<List<int>>();
+    var fileExtension = '';
+
+    switch (config.encoder) {
```

Comment:
WAVE can't be streamed with headers. Please remove it.
But maybe we can enable PCM recording this way?